src/cargo/core/compiler/custom_build.rs

```diff
@@ -401,10 +400,28 @@ fn build_work(cx: &mut Context<'_, '_>, unit: &Unit) -> CargoResult<Job> {
         let mut diagnostics = Vec::new();
         let script_result = cmd.exec_with_streaming(
             &mut |stdout| {
-                if let Some(warning) = stdout.strip_prefix(CARGO_WARNING) {
-                    diagnostics.push(Diagnostic::Warning(warning.to_owned()));
-                } else if let Some(error) = stdout.strip_prefix(CARGO_ERROR) {
-                    diagnostics.push(Diagnostic::Error(error.to_owned()));
+                let kv = stdout.strip_prefix(CARGO_LINE_PREFIX).and_then(|kv| {
+                    let mut kv = kv.splitn(2, '=');
+                    Some((kv.next()?, kv.next()?))
+                });
+                if let Some((key, value)) = kv {
+                    match key {
+                        "warning" => diagnostics.push(Diagnostic::Warning(value.to_owned())),
+                        "error" => diagnostics.push(Diagnostic::Error(value.to_owned())),
+                        "warning+" => {
```

Comment:
Can we split this out into a separate Issue/PR so we can  make sure we are agreed on the semantics we want for this before going to the review stage and so we don't block your other work on it?

Reply:
Do you mean just the `+=` syntax?

Reply:
Yes, `+=` syntax.  While discussing alternatives can be interesting, again I feel its best that we have a dedicated discussion on this, so please split this out.  This is not strictly required for merging error messages.

Reply:
This was discussed in the cargo team meeting.  We came up with several concerns and it would be best to have a dedicated Issue for finishing the conversation on them.